CTNotificationContent/Templates/Carousel/Controller/CTCarouselController.swift

```diff
@@ -125,6 +144,31 @@ import UserNotificationsUI
         }
     }
     
+    @objc public override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
```

Comment:
`traitCollectionDidChange` is deprecated from iOS 17, use the new protocol API at every places for iOS 17 and above

Reply:
Done